src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,153 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { div, h, h1, img } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, ButtonSecondary } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { MarkdownViewer, newWindowLinkRenderer } from 'src/components/markdown';
+import scienceBackground from 'src/images/science-background.jpg';
+import { Ajax } from 'src/libs/ajax';
+import { signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useOnMount, useStore } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState<boolean>();
+  const { signInStatus, termsOfService } = useStore(authStore);
+  const acceptedLatestTos = signInStatus === 'signedIn' && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = signInStatus === 'signedIn' && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState<string>();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const { enabled } = await Ajax().User.acceptTos();
+      if (enabled) {
+        const termsOfService = await Ajax().User.getTermsOfServiceComplianceStatus();
+
+        const registrationStatus = 'registered';
+        authStore.update((state) => ({ ...state, registrationStatus, termsOfService }));
+        Nav.goToPath('root');
+      } else {
+        reportError(
+          'Error accepting terms of service, unexpected backend error occurred.',
+          new Error('Unexpected backend error')
+        );
+      }
+    } catch (error) {
+      reportError('Error accepting Terms of Service', error);
+    } finally {
+      setBusy(false);
+    }
+  };
+
+  const reject = async () => {
+    try {
+      setBusy(true);
+      await Ajax().User.rejectTos();
+    } catch (error) {
+      reportError('Error rejecting Terms of Service', error);
+    } finally {
+      setBusy(false);
+      signOut('declinedTos');
+    }
+  };
+
+  const continueButton = () => {
```

Comment:
I know this is copied over and thus isn't really in scope for this PR, but this could use a better name. `continueButton` makes me think this is a button element or something like that, not an action/event callback.

```suggestion
  const onContinueUnderGracePeriod = () => {
```